desc/objectives/_generic.py

```diff
@@ -144,35 +146,32 @@ def get_vars(fun):
 
         self._dim_f = jax.eval_shape(self._fun_wrapped, dummy_data).size
         self._scalar = self._dim_f == 1
-        self._args = get_params(
-            self._data_keys,
-            obj="desc.equilibrium.equilibrium.Equilibrium",
-            has_axis=grid.axis.size,
-        )
         profiles = get_profiles(self._data_keys, obj=eq, grid=grid)
         transforms = get_transforms(self._data_keys, obj=eq, grid=grid)
         self._constants = {
             "transforms": transforms,
             "profiles": profiles,
         }
 
-        super().build(eq=eq, use_jit=use_jit, verbose=verbose)
+        super().build(things=eq, use_jit=use_jit, verbose=verbose)
 
-    def compute(self, *args, **kwargs):
+    def compute(self, params, constants=None):
         """Compute the quantity.
 
         Parameters
         ----------
-        args : ndarray
-            Parameters given by self.args.
+        params : dict
```

Comment:
does it have to be Equilibrium DOFs?

Reply:
for now, this objective still assumes you're optimizing an equilibrium. We could probably generalize it somehow?